datalad/support/gitrepo.py

```diff
@@ -3115,7 +3116,7 @@ def get_tracking_branch(self, branch=None, remote_only=False):
             (remote or None, refspec or None) of the tracking branch
         """
         if branch is None:
-            branch = self.get_active_branch()
+            branch = self.get_corresponding_branch() or self.get_active_branch()
```

Comment:
Hmm, this change isn't needed for the test case, as `AnnexRepo.get_tracking_branch(..., branch=None, corresponding=True)` already handles it.  And if GitRepo is being used directly, `get_corresponding_branch` always returns None.

Is the idea with this change just that if `GitRepo.get_corresponding_branch` ever changed to do anything meaningful, this is one (of many) spots that should take it into account?

Reply:
Ah, this might be an artifact of the order in which I explored the situation. I will investigate and adjust accordingly. Thx!

Reply:
Remove, thx for the catch!